src/main/java/com/google/devtools/build/lib/actions/cache/VirtualActionInput.java

```diff
@@ -67,6 +67,11 @@ public byte[] atomicallyWriteRelativeTo(Path execRoot, String uniqueSuffix) thro
   protected byte[] atomicallyWriteTo(Path outputPath, String uniqueSuffix) throws IOException {
     Path tmpPath = outputPath.getFileSystem().getPath(outputPath.getPathString() + uniqueSuffix);
     tmpPath.getParentDirectory().createDirectoryAndParents();
+    try {
+      tmpPath.delete();
+    } catch (IOException e) {
+      // Ignore.
```

Comment:
vfs.Path is defined not to throw if deletion failed because the file didn't exist - instead you should check the return value if you care: https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/vfs/Path.java;l=584;drc=666fce514b87e6901b033f3399e2d4b56a856429. I haven't checked all FileSystem implementations, but at least the Unix one agrees.

Reply:
Ah, nice. That means that we can call `tmpPath.delete()` without catching any exceptions. We don't care whether the file existed or not. I just simplified it. Thanks!